cdap-app-templates/cdap-etl/cdap-etl-lib/src/test/java/co/cask/cdap/template/etl/realtime/source/SqsSourceTest.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.template.etl.realtime.source;
+
+import co.cask.cdap.api.data.format.StructuredRecord;
+import co.cask.cdap.common.utils.Networks;
+import co.cask.cdap.template.etl.api.realtime.SourceState;
+import co.cask.cdap.template.etl.common.MockEmitter;
+import co.cask.cdap.template.etl.common.MockRealtimeContext;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.services.sqs.AmazonSQSClient;
+import org.elasticmq.Node;
+import org.elasticmq.NodeAddress;
+import org.elasticmq.rest.RestServer;
+import org.elasticmq.rest.sqs.SQSRestServerBuilder;
+import org.elasticmq.storage.inmemory.InMemoryStorage;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+public class SqsSourceTest {
+  private static final int ELASTIMQ_PORT = Networks.getRandomPort();
+  private RestServer sqsServer;
+  private Node elasticNode;
+  private static final String HOST_URL = "http://localhost:" + ELASTIMQ_PORT;
```

Comment:
Take a look at ``BatchETLDBAdapterTest``, to see how you can encapsulate these connection params in to a Server class.